### PR TITLE
cmd/snap: display a link to data privacy notice for interactive snap login

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

File: cmd/snap/cmd_login.go [hunk 1]
```diff
@@ -113,6 +113,7 @@ func (x *cmdLogin) Execute(args []string) error {
 
 	email := x.Positional.Email
 	if email == "" {
+		fmt.Fprint(Stdout, i18n.G("Authenticated operation on this device will be under terms of service found at: https://www.ubuntu.com/legal/<some-snap-store-url>")+"\n\n")
```
Comment (zyga) [inline cmd/snap/cmd_login.go]: I'd recommend moving the URL to somewhere under snapcraft.io

**Comment by pedronis** [comment]
marking blocked until we have final text/url

**Comment by pedronis** [comment]
should we mention the URL also in the help text of the command?

**Comment by sparkiegeek** [comment]
> should we mention the URL also in the help text of the command?

(IANAL) I don't think that's strictly necessary - the key issue this is addressing is letting users know at the point of capture what we're doing with their email address and other data associated with it.

File: cmd/snap/cmd_login.go (hunk 1 above)
Comment (sparkiegeek) [inline cmd/snap/cmd_login.go]: We should move this outside of the `if` -this needs to be presented in the case of their email being given on the CLI too.

Comment (chipaca) [reply cmd/snap/cmd_login.go]: why only prompt when email wasn't given?

Comment (zyga) [reply cmd/snap/cmd_login.go]: @sparkiegeek who operates snapcraft.io?

File: cmd/snap/cmd_login.go [hunk 2]
```diff
@@ -111,6 +111,8 @@ func (x *cmdLogin) Execute(args []string) error {
 		return ErrExtraArgs
 	}
 
+	fmt.Fprint(Stdout, i18n.G("Authenticated operation on this device will be under terms of service found at: https://www.ubuntu.com/legal/<some-snap-store-url>")+"\n\n")
```
Comment (sparkiegeek) [inline cmd/snap/cmd_login.go]: I haven't heard an update on wording but the URL is https://www.ubuntu.com/legal/dataprivacy/snap-store

I'll follow up

File: cmd/snap/cmd_login.go [hunk 3]
```diff
@@ -111,6 +111,8 @@ func (x *cmdLogin) Execute(args []string) error {
 		return ErrExtraArgs
 	}
 
+	fmt.Fprint(Stdout, i18n.G("Authenticated operation on this device will be under terms of service found at: https://www.ubuntu.com/legal/dataprivacy/snap-store")+"\n\n")
```
Comment (sparkiegeek) [inline cmd/snap/cmd_login.go]: "Personal information is handled as per our privacy notice at https://www.ubuntu.com/legal/dataprivacy/snap-store"

^ has been +1'd by Legal, let's do it!

File: cmd/snap/cmd_login_test.go [hunk 4]
```diff
@@ -57,9 +58,9 @@ func (s *SnapSuite) TestLoginSimple(c *C) {
 	rest, err := snap.Parser().ParseArgs([]string{"login", "foo@example.com"})
 	c.Assert(err, IsNil)
 	c.Assert(rest, DeepEquals, []string{})
-	c.Check(s.Stdout(), Equals, `Password of "foo@example.com": 
+	c.Check(strings.HasSuffix(s.Stdout(), `Password of "foo@example.com":
```
Comment (mvo5) [inline cmd/snap/cmd_login_test.go]: It seems we should also check the output contains our legal url, no?

File: cmd/snap/cmd_login.go [hunk 5]
```diff
@@ -111,6 +111,10 @@ func (x *cmdLogin) Execute(args []string) error {
 		return ErrExtraArgs
 	}
 
+	//TRANSLATORS: after the "... at" follows a URL in the next line
+	fmt.Fprint(Stdout, i18n.G("Personal information is handled as per our privacy notice at\n"))
+	fmt.Fprintf(Stdout, "https://www.ubuntu.com/legal/dataprivacy/snap-store\n\n")
```
Comment (pedronis) [inline cmd/snap/cmd_login.go]: this can also be a Fprint, no?

**Pull request merged**